app/javascript/components/lease_ups/LeaseUpsTableContainer.js

```diff
@@ -1,10 +1,90 @@
 import React from 'react'
-import _ from 'lodash'
+import { set, cloneDeep, trim, findIndex, map } from 'lodash'
+import moment from 'moment'
 
+import apiService from '~/apiService'
 import LeaseUpsTable from './LeaseUpsTable'
-import appPaths from '~/components/appPaths'
+import LeaseUpsStatusModalWrapper from './LeaseUpsStatusModalWrapper'
+import appPaths from '~/utils/appPaths'
 
 class LeaseUpTableContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      statusModal: {
+        isOpen: false,
+        status: null,
+        applicationId: null,
+        showAlert: null,
+        loading: false,
+      },
+      results: this.props.results
+    }
+  }
+
+  updateState = (path, value) => {
+    this.setState(prevState => {
+      var newState = cloneDeep(prevState)
```

Comment:
what are we deep cloning the state? whatever we return in it gets merged to the current state. so if you state is  `{a:1, b2}` and you return `{b:3}`. The state becomes `{a1, b:3}`. you do not lose the other keys.
also passing a function to setState is technically used only if you need a value from the `prevState` because react updates the state async. Ping me on slack if this is confusing.

Reply:
my aim here was to not override nested objects - i'll ping you on slack to discuss further - definitely would be create if we don't have to deep clone!

Reply:
@fddayan i pushed the new state update functions we worked on